### ISSUE TITLE
Application of templates/styles to portions of the site

### ISSUE THREAD

**Issue by guardrex** [dotnet/docfx#2621]
We can't apply templates by page/folder? I'd like to apply a template (the styles specifically) selectively to the `index.md` file at the root but apply a different template to the rest of the pages/folders in the site.

It looks like your preferred approach is to handle this scenario within the CSS to make classes for the root `index.md` specific to that root `index.md` page. Not simple to work out for things like Bootstrap `container` and `article`, since these classes are used outside of the `{{!body}}` in the template file.

I tried an approach where I placed a different `docfx.json` in a folder thinking that it could overwrite the `template` node to only apply the `default` template to the markdown files of that folder. This approach didn't work.

TL;DR Can I apply different templates (thus different stylesheets) to different pages/folders in the site?

cc/ @vicancy @superyyrrzz

**Comment by guardrex** [comment]
I'll add my hack to make it work ...

I put a stylesheet (`main.css`) into a `stylesheets` folder in the root. I reference that stylesheet only from the root `index.md` page ...

```html
<link rel="stylesheet" type="text/css" href="stylesheets/main.css" />
```

It works well enough.

**Comment by superyyrrzz** [comment]
DocFX's [homepage](https://raw.githubusercontent.com/dotnet/docfx/dev/Documentation/index.md) specifies `documentType` to `index`, to use the custom template [index.html.tmpl](https://github.com/dotnet/docfx/blob/dev/Documentation/custom_template/index.html.tmpl). Is this what you want?

**Comment by guardrex** [comment]
How do these paths work? Your docs are not clear ...

```
{{!include('styles/master.css')}}
```
No error on build, but no styles are applied either.

**Comment by guardrex** [comment]
The templating system is very difficult to understand and the docs aren't helpful. I'm going to go back to my workaround, which works and is easy to understand and use.

**Comment by maccettura** [comment]
@superyyrrzz I am venturing down a similar path as OP now and I was hoping you could clarify your comment.  I see that I can add the `documentType` to the frontmatter of a `.md` file, but I am unable to figure out how to get a custom template to render.

Basically I have an API "section" that is configured in my docfx.json and I want that _specific_ section to have a very customized template (it basically strips out everything except properties).  For the other API's I would like to have a different template.  Is there any documentation you would recommend for my problem?  I am having a hard time figuring it out with the docfc documentation.

**Comment by superyyrrzz** [comment]
@maccettura  `documentType` is a feature specific for `.md` input. For API pages generated from source code, do you mean to differentiate them using different layout? If so, you can custom the existing template `ManagedReference.html.primary.tmpl` to have some conditional layouts. Of course, you need a way to tell template when to use which style. An option is to use `fileMetadata` to assign different values for you want.

Doc: https://dotnet.github.io/docfx/tutorial/howto_create_custom_template.html

**Comment by maccettura** [comment]
@superyyrrzz I have a project that solely contains blocks and pages for an Episerver CMS project.  I want to have some very highly customized and "skinny" documentation for that project only.  There are other projects in the solution that will use the regular default template.  So I basically need a way to apply a specific template to one API section.  I hesitate to modify the `ManagedReference.html.primary.tmpl` because I think that would effect the "normal" sections I wish to render, right?

**Comment by superyyrrzz** [comment]
@maccettura Got your ideas. If the Episerver project's layout does not share much with the default one for API, customizing `ManagedReference.html.primary.tmpl` may not be a good choice.

Maybe you can [create a document processor](https://dotnet.github.io/docfx/tutorial/howto_build_your_own_type_of_documentation_with_custom_plug-in.html#create-a-document-processor) inherited from ManagedReferencedDocumentProcessor for this particular project files. Different document processor can have different templates.
